domains/Discussions/Tests/Feature/QuestionsDeleteTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Domains\Discussions\Tests\Feature;
+
+use Domains\Accounts\Database\Factories\UserFactory;
+use Domains\Accounts\Enums\AccountTypeEnum;
+use Domains\Accounts\Models\User;
+use Domains\Discussions\Database\Factories\QuestionFactory;
+use Domains\Discussions\Models\Question;
+use Faker\Factory;
+use Faker\Generator;
+use Illuminate\Http\Response;
+use Illuminate\Support\Carbon;
+use Laravel\Lumen\Testing\DatabaseMigrations;
+use Tests\TestCase;
+
+class QuestionsUpdateTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    private Generator $faker;
+    private User $user;
+    private Question $question;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->faker = Factory::create();
+        $this->user = UserFactory::new()->create();
+        $this->question = QuestionFactory::new(['author_id' => $this->user->id])->create();
+    }
+
+    /** @test */
+    public function it_soft_deletes_a_question_i_own(): void
+    {
+        Carbon::setTestNow();
+
+        $response = $this->actingAs($this->user)
+            ->delete(route('discussions.questions.delete', ['questionId' => $this->question->id]));
+
+        $this->assertResponseStatus(Response::HTTP_NO_CONTENT);
+        $this->assertTrue($response->response->isEmpty());
+        $this->seeInDatabase('questions', [
+            'id' => $this->question->id,
+            'updated_at' => Carbon::now(),
+            'deleted_at' => Carbon::now(),
```

Comment:
We need to "freeze" time in order to avoid possible intermittent errors due to execution happening during a second's change in the timeframe.

Reply:
True, moved to `setUp()`